cypress/integration/chi-vue/pagination.spec.js

```diff
@@ -0,0 +1,414 @@
+            .find(`.${PAGINATION_CLASSES.PAGE_SIZE} select`)
+            .as('pageSelect')
+            .select('40')
+            .then(() => {
+              expect(spy).to.be.calledOnce;
+            });
+          cy.get(`@pageSelect`)
+            .select('20')
+            .then(() => {
+              expect(spy).to.be.calledTwice;
+            });
+        });
+    });
+
+    it('should change page size on select change', () => {
+      cy.get(`[data-cy='pagination-page-size']`)
+        .find(`.${PAGINATION_CLASSES.PAGE_SIZE} select`)
+        .as('pageSelect')
+        .should('contain', '20');
+      cy.get('@pageSelect')
+        .select('40')
+        .then(() => {
+          cy.get('@pageSelect').should('contain', '40');
+        });
+      cy.get(`@pageSelect`)
+        .select('20')
+        .then(() => {
+          cy.get('@pageSelect').should('contain', '20');
+        });
+    });
+  });
+
+  describe('page jumper', () => {
+    it('should show a page jumper', () => {
+      const selectors = [
+        `[data-cy='pagination-page-jumper']`,
+        `[data-cy='pagination-page-jumper-inverse']`
+      ];
+
+      selectors.forEach(sel => {
+        cy.get(`${sel}`)
+          .find(`.${PAGINATION_CLASSES.JUMPER}`)
+          .as('pageJumper')
+          .should('contain', 'Go to page:');
+        cy.get('@pageJumper')
+          .find('input')
+          .type('2')
+          .should('have.value', '2');
+      });
+    });
+  });
+
+  describe('compact', () => {
+    describe('base', () => {
+      it(`should have class .${PAGINATION_CLASSES.COMPACT}`, () => {
+        const selectors = [
+          `[data-cy='pagination-compact-base']`,
+          `[data-cy='pagination-compact-page-jumper-inverse']`
+        ];
+
+        selectors.forEach(sel => {
+          cy.get(`${sel}`).should('have.class', PAGINATION_CLASSES.COMPACT);
+        });
+      });
+    });
+
+    describe('page jumper', () => {
+      it('should show an inline page jumper', () => {
+        const selectors = [
+          `[data-cy='pagination-compact-page-jumper']`,
+          `[data-cy='pagination-compact-page-jumper-inverse']`
+        ];
+
+        selectors.forEach(sel => {
+          cy.get(`${sel}`)
+            .find(`.${PAGINATION_CLASSES.JUMPER}`)
+            .as('pageJumper')
+            .should('contain', 'of');
+          cy.get('@pageJumper').should('contain', 3);
+          cy.get('@pageJumper')
+            .find('input')
+            .type('{backspace}3')
+            .should('have.value', '3');
+        });
+      });
+    });
+
+    describe('first and last page buttons', () => {
+      it('should show first and last page buttons', () => {
+        const selectors = [
+          `[data-cy='pagination-compact-first-last']`,
+          `[data-cy='pagination-compact-first-last-inverse']`
+        ];
+
+        selectors.forEach(sel => {
+          cy.get(`${sel}`)
+            .find("[data-page='1']")
```

Comment:
This will take into account the first page (<<) but also the page 1 button.
Could you find a found to specifically select the First Page button, please?
Keep in mind, that it comes with a special icon that you could use as a selector.

Reply:
oh! .icon-page-first and .icon-page-last you mean?